packages/components/src/internal/components/chart/Chart.tsx

```diff
@@ -13,193 +13,186 @@
+    const [loadError, setLoadError] = useState<string>(undefined);
+    const filterKey = useMemo(() => computeFilterKey(filters), [filters]);
+    const loadReport = useCallback(async () => {
+        setLoadingState(LoadingState.LOADING);
+        setLoadError(undefined);
+
+        try {
+            const html = await api.fetchRReport(reportId, filters);
+            setReportHtml(html);
+        } catch (e) {
+            setLoadError(e.exception);
+        } finally {
+            setLoadingState(LoadingState.LOADED);
+        }
+        // We purposely don't use filters as a dep, see note in computeFilterKey
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [api, reportId, filterKey]);
+    const imageUrls = useMemo(() => {
+        if (reportHtml !== undefined) {
+            // The HTML returned by our server includes a bunch of stuff we don't want. So instead of inserting it
+            // directly we'll just grab the URLs for all the images named "resultImage", which are the outputs from an
+            // R Report.
+            const el = document.createElement('div');
+            el.innerHTML = reportHtml;
+            const resultImages = el.querySelectorAll('img[name="resultImage"]');
+            return Array.from(resultImages).map((img: HTMLImageElement) => img.src);
+        }
 
-function getVisualizationConfig(reportId: string): Promise<VisualizationConfigModel> {
-    return new Promise((resolve, reject) => {
-        Query.Visualization.get({
-            reportId,
-            name: undefined,
-            schemaName: undefined,
-            queryName: undefined,
-            success: response => {
-                resolve(VisualizationConfigModel.create(response.visualizationConfig));
-            },
-            failure: reject,
-        });
-    });
-}
+        return undefined;
+    }, [reportHtml]);
 
-class VisualizationConfigModel extends Record({
-    queryConfig: undefined,
-    chartConfig: undefined,
-}) {
-    declare queryConfig: QueryConfigModel;
-    declare chartConfig: ChartConfigModel;
-
-    static create(raw: any): VisualizationConfigModel {
-        return new VisualizationConfigModel(
-            Object.assign({}, raw, {
-                chartConfig: new ChartConfigModel(raw.chartConfig),
-                queryConfig: new QueryConfigModel(raw.queryConfig),
-            })
-        );
+    useEffect(() => {
+        if (error) {
+            // We won't bother loading anything if the chart object has an error
+            setLoadingState(LoadingState.LOADED);
+        } else {
+            loadReport();
+        }
+    }, [error, loadReport]);
+
+    return (
+        <div className="r-report">
+            {isLoading(loadingState) && <LoadingSpinner />}
+            {error !== undefined && <span className="text-danger">{error}</span>}
+            {loadError !== undefined && <span className="text-danger">{loadError}</span>}
+            {imageUrls !== undefined && (
```

Comment:
if this is an R report that doesn't render any images, will this just result in a blank page/modal? Should we add some sort of warning or alert message saying that the report was at least attempted but didn't include any images?

Reply:
I think that is something we can address later. For now I am going to keep it simple because this has such a narrow use case.